src/system_data.jl

```diff
@@ -311,6 +311,21 @@ function clear_time_series!(data::SystemData)
     reset_info!(data.time_series_params)
 end
 
+function clear_time_series_transformation!(data::SystemData)
```

Comment:
If the system has other forecasts (Deterministic, Probabilistic, or Scenarios) then bad things will happen. Calling this function will cause all of their metadata to get deleted. The time series storage (HDF) will still contain the raw data.

You could solve that problem by adding a check for the existence of only SingleTimeSeries and DeterministicSingleTimeSeries.

When we discussed this problem this week I was thinking about a more generic approach:
```
function remove_time_series!(data::SystemData, ::Type{T}) where T <: TimeSeriesData
    for component in iterate_components_with_time_series(data.components)
        for ts in get_time_series_multiple(component, type = T)
            remove_time_series!(data, typeof(ts), component, get_name(ts))
        end
    end
end
```
Example usage: `remove_time_series!(sys, DeterministicSingleTimeSeries)`

I tested this and found an unrelated problem. We need this addition:
```
get_name(value::DeterministicSingleTimeSeries) = get_name(value.single_time_series)
```
The generic solution also needs better handling of resetting of the time_series_params. The current code doesn't handle removing all time series one-by-one. For now we could do what you have here, but with a check. A better fix would be to handle it at a lower level.

I'm fine with either approach.